akka-stream/src/main/resources/reference.conf

```diff
@@ -45,6 +45,16 @@ akka {
       # this may cause an initial runtime overhead, but most of the time fusing is
       # desirable since it reduces the number of Actors that are created.
       auto-fusing = on
+      
+      # Those stream elements which have explicit buffers (like mapAsync, mapAsyncUnordered,
+      # buffer, flatMapMerge, Source.actorRef, Source.queue, etc.) will preallocate a fixed
+      # buffer upon stream materialization if the requested buffer size is less than this
+      # configuration parameter. The default is very high because failing early is better
+      # than failing under load.
+      #
+      # Buffers larger than this value will use a dynamic approach that has a higher
+      # per-element cost.
+      max-fixed-buffer-size = 1000000000
```

Comment:
What's the effect on performance going from monomorphic to bimorphic given
the call site pollution the interpreter causes?

-- 
Cheers,
√
On Feb 8, 2016 9:27 AM, "Roland Kuhn" <notifications@github.com> wrote:

> In akka-stream/src/main/resources/reference.conf
> <https://github.com/akka/akka/pull/19703#discussion_r52141626>:
>
> > @@ -45,6 +45,16 @@ akka {
> >        # this may cause an initial runtime overhead, but most of the time fusing is
> >        # desirable since it reduces the number of Actors that are created.
> >        auto-fusing = on
> > +
> > +      # Those stream elements which have explicit buffers (like mapAsync, mapAsyncUnordered,
> > +      # buffer, flatMapMerge, Source.actorRef, Source.queue, etc.) will preallocate a fixed
> > +      # buffer upon stream materialization if the requested buffer size is less than this
> > +      # configuration parameter. The default is very high because failing early is better
> > +      # than failing under load.
> > +      #
> > +      # Buffers larger than this value will use a dynamic approach that has a higher
> > +      # per-element cost.
> > +      max-fixed-buffer-size = 1000000000
>
> I take that to imply that the feature is good? (there was no discussion
> yet, so I just threw the proposal into the ring)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/19703/files#r52141626>.
>